src/main/java/org/mockito/internal/stubbing/answers/DefaultValidableAnswer.java

```diff
@@ -0,0 +1,34 @@
+/*
+ * Copyright (c) 2007 Mockito contributors
+ * This program is made available under the terms of the MIT License.
+ */
+
+package org.mockito.internal.stubbing.answers;
+
+import org.mockito.invocation.InvocationOnMock;
+import org.mockito.stubbing.Answer;
+
+import static org.mockito.internal.exceptions.Reporter.wrongTypeReturnedByDefaultAnswer;
+
+public class DefaultValidableAnswer {
```

Comment:
Ah I see! I think it is possible to seperate the concerns here, if you keep the parameters of the "old" validateDefaultAnswerReturnedValue method, something like that:

```java
static void validatedReturnValue(InvocationOnMock invocation, Object actualReturnValue) 
```

IMO this would read a bit nicer on the call side:

```java
Object returnValue = answer.answer(invocation);
validatedReturnValue(invocation, returnValue);
```

vs.

```java
Object ret = new DefaultValidableAnswer(mockSettings.getDefaultAnswer()).validatedAnswerFor(invocation);  
//     ^ for a reader it may not be clear that this is the return value of the answer
```